tests/unit/TechRecordsService.unitTest.ts

```diff
@@ -759,6 +757,262 @@ describe("updateTechRecord", () => {
+              await techRecordsService.updateAttributesOutsideTechRecordsArray(techRecord, payload);
+              expect(techRecord.primaryVrm).toEqual("ABCD943");
+              expect(techRecord.secondaryVrms.length).toEqual(2);
+              expect(techRecord.secondaryVrms).toContain("LKJH654");
+              expect(payload.techRecord[0].reasonForCreation).toEqual(`VRM updated from LKJH654 to ABCD943. Updated VRM`);
+            });
+
+            context("and the vehicle didn't have VRMs", () => {
+              it("should not append undefined into secondaryVrms array", async () => {
+                const techRecord: any = cloneDeep(records[43]);
+                techRecord.vrms = [];
+                delete techRecord.secondaryVrms;
+                delete techRecord.primaryVrm;
+                const MockDAO = jest.fn().mockImplementation(() => {
+                  return {
+                    getBySearchTerm: () => {
+                      return Promise.resolve({
+                        Items: [],
+                        Count: 0,
+                        ScannedCount: 1
+                      });
+                    }
+                  };
+                });
+                const techRecordsService = new TechRecordsService(new MockDAO());
+                const payload: any = cloneDeep(techRecord);
+                payload.techRecord[0].reasonForCreation = "Updated VRM";
+                payload.primaryVrm = "ABCD943";
+                await techRecordsService.updateAttributesOutsideTechRecordsArray(techRecord, payload);
+                expect(techRecord.primaryVrm).toEqual("ABCD943");
+                expect(techRecord.secondaryVrms.length).toEqual(0);
+              });
+            });
+          });
+          context("and the primaryVrm is already present on another vehicle", () => {
+            it("should return Error 400 primaryVrm already exists", async () => {
+              const techRecord: any = cloneDeep(records[43]);
+              techRecord.vrms = [{isPrimary: true, vrm: "LKJH654"}];
+              const MockDAO = jest.fn().mockImplementation(() => {
+                return {
+                  getBySearchTerm: () => {
+                    return Promise.resolve({
+                      Items: [techRecord],
+                      Count: 1,
+                      ScannedCount: 1
+                    });
+                  }
+                };
+              });
+              const techRecordsService = new TechRecordsService(new MockDAO());
+              const payload: any = cloneDeep(records[43]);
+              payload.primaryVrm = "ABCD943";
+              try {
+                expect(await techRecordsService.updateAttributesOutsideTechRecordsArray(techRecord, payload)).toThrowError();
+              } catch (errorResponse) {
+                expect(errorResponse.statusCode).toEqual(400);
+                expect(errorResponse.body).toEqual("Primary VRM ABCD943 already exists");
```

Comment:
I was thinking that we have a similar error message pattern instead of using strict matching we could be more permissive and use toMatch matcher with a regex pattern. In this case we could have a regex. Do we know all the error messages we may have already? I can't remember seeing such a document so I wonder whether it will cause more cognitive overhead than anything in fact.